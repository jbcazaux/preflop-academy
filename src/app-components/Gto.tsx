import { useEffect, useState } from 'react'
import gto from 'data/gto-client'
import Action from 'domain/action'
import Hand from 'domain/hand'
import Position from 'domain/position'

import ActionComponent from 'components/Action'

interface Props {
  hero: Position
  actions: ReadonlyArray<Action>
  hand: Hand
}

const Gto = ({ hero, actions, hand }: Props) => {
  const [move, setMove] = useState<string>('')

  useEffect(() => {
    gto(
      hero,
      actions.map(a => a.position),
      hand
    ).then(result => {
      setMove(result || 'N/A')
    })
  }, [hero, actions, hand])

  return <ActionComponent>You should : {move}</ActionComponent>
}

export default Gto
