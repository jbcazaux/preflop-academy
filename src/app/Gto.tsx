import Hand from 'domain/hand'
import gto from 'data/gto'
import Position from 'domain/position'
import ActionComponent from 'components/Action'
import Action from 'domain/action'

interface Props {
  hero: Position
  actions: ReadonlyArray<Action>
  hand: Hand
}

const Gto: React.FC<Props> = ({ hero, actions, hand }) => {
  const move = gto(
    hero,
    actions.map(a => a.position),
    hand
  )
  return <ActionComponent>You should : {move || 'N/A'}</ActionComponent>
}

export default Gto
