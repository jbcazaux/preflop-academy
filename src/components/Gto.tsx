import Hand from 'domain/hand'
import React from 'react'
import gto from '../data/gto'
import styled from 'styled-components'

const Advice = styled.div`
  display: flex;
  margin: 10px auto;
  font-weight: bold;
  font-size: 2em;
`

interface Props {
  buttonPostion: number
  raisePositions: ReadonlyArray<number>
  hand: Hand
}

const Gto: React.FC<Props> = ({ buttonPostion, raisePositions, hand }) => {
  const advice = gto(buttonPostion, raisePositions, hand)
  return <Advice>{advice}</Advice>
}

export default Gto
