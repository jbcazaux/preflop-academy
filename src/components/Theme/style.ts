import { DefaultTheme } from 'styled-components'

const style: DefaultTheme = {
  borderRadius: '5px',
  breakpoints: {
    mobile: 420,
    tablet: 768,
    desktop: 1200,
    min: {
      mobile: 'min-width: 420px',
      tablet: 'min-width: 768px',
      desktop: 'min-width: 1200px',
    },
    max: {
      mobile: 'max-width: 420px',
      tablet: 'max-width: 768px',
      desktop: 'max-width: 1200px',
    },
  },
  colors: {
    primary: '#93B7BE',
    secondary: '#F7D488',
    background: '#F7F1F7',
    black: '#222',
    white: '#D5E2E3',
    inHand: 'red',
    onBoard: 'red',
    buttons: {
      default: '#D5E2E3',
    },
    deck: {
      diamond: 'red',
      heart: 'red',
      club: 'black',
      spade: 'black',
    },
    loader: {
      background: '#D5E2E3',
      color: '#3498DB',
    },
    menu: {
      menu1: '#de004e',
      menu2: '#860029',
      menu3: '#321450',
      menu4: '#d40078',
      menu5: '#f6019d',
    },
    range: {
      active: '#FE5F55',
      suited: '#2DDAEE',
      offsuit: 'pink',
      selected: '#d40078',
    },
    table: {
      board: '#58BD86',
      border: '#4C4949',
      button: '#E8F63C',
      action: '#de004e',
    },
  },
}

export default style
