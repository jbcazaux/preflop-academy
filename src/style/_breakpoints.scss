/**
 breakpoints: {
    mobile: 420,
    tablet: 768,
    desktop: 1200,
    min: {
      mobile: 'min-width: 420px',
      tablet: 'min-width: 768px',
      desktop: 'min-width: 1200px',
    },
    max: {
      mobile: 'max-width: 420px',
      tablet: 'max-width: 768px',
      desktop: 'max-width: 1200px',
    },
  },

**/

$breakpoints-min-mobile: 420px;
$breakpoints-max-mobile: 420px;
$breakpoints-min-tablet: 768px;
$breakpoints-max-tablet: 768px;
$breakpoints-min-desktop: 1200px;
$breakpoints-max-desktop: 1200px;
