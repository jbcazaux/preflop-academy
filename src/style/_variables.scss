$primary: #93b7be;
$secondary: #f7d488;
$background: #f7f1f7;
$black: #222;
$white: #d5e2e3;
$inHand: red;
$onBoard: red;
$button-default: #d5e2e3;

$deck-diamond: red;
$deck-heart: red;
$deck-club: black;
$deck-spade: black;

$loader-background: #d5e2e3;
$loader-color: #3498db;

$menu1: #de004e;
$menu2: #860029;
$menu3: #321450;
$menu4: #d40078;
$menu5: #f6019d;

$range-active: #fe5f55;
$range-suited: #2ddaee;
$range-offsuit: pink;
$range-selected: #d40078;

$table-board: #58bd86;
$table-border: #4c4949;
$table-button: #e8f63c;
$table-action: #de004e;

:export {
  menu1: #de004e;
  menu2: $menu2;
  menu3: $menu3;
  menu4: $menu4;
  menu5: $menu5;
  background: $background;
  deckHeart: $deck-heart;
  deckDiamond: $deck-diamond;
  deckClub: $deck-club;
  deckSpade: $deck-spade;
}
