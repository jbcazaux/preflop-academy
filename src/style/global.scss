body,
html,
* {
  padding: 0;
  margin: 0;
}

html,
body {
  font-family: Calibri, 'Helvetica Neue', Arial, sans-serif;
  width: 100%;
  height: 100%;
}

* {
  box-sizing: border-box;
}

html {
  scroll-behavior: smooth;
}
