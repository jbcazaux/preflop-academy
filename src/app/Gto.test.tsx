import { render } from '@testing-library/react'
import Gto from 'app/Gto'
import Hand from 'domain/hand'
import 'jest-styled-components'
import Position from 'domain/position'
import Move from 'domain/move'
import { ThemeProvider } from 'styled-components'
import style from 'components/Theme/style'

jest.mock('data/gto', () => () => Move.OPEN)
describe('Gto', () => {
  it('displays whatever gto says', () => {
    const { container } = render(
      <ThemeProvider theme={style}>
        <Gto hero={Position.B} actions={[]} hand={Hand.newHand} />
      </ThemeProvider>
    )

    expect(container).toMatchSnapshot()
  })
})
