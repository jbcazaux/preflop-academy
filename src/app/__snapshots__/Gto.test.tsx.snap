// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gto displays whatever gto says 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 10px 0;
  font-weight: bold;
  font-size: 2em;
}

@media (max-width:1200px) {
  .c0 {
    font-size: 1em;
  }
}

<div>
  <div
    class="c0"
  >
    You should : 
    OPEN
  </div>
</div>
`;
